extensions/funqy/funqy-http/runtime/src/main/java/io/quarkus/funqy/runtime/bindings/http/VertxRequestHandler.java

```diff
@@ -98,23 +98,22 @@ public void handle(RoutingContext routingContext) {
                 dispatch(routingContext, invoker, finalInput);
             });
         } else if (routingContext.request().method() == HttpMethod.POST) {
-            routingContext.request().bodyHandler(buff -> {
-                Object input = null;
-                if (buff.length() > 0) {
-                    ByteBufInputStream in = new ByteBufInputStream(buff.getByteBuf());
-                    ObjectReader reader = (ObjectReader) invoker.getBindingContext().get(ObjectReader.class.getName());
-                    try {
-                        input = reader.readValue((InputStream) in);
-                    } catch (Exception e) {
-                        log.error("Failed to unmarshal input", e);
-                        routingContext.fail(400);
-                        return;
-                    }
+            var buff = routingContext.getBody();
+            Object input = null;
+            if (buff != null && buff.length() > 0) {
+                ByteBufInputStream in = new ByteBufInputStream(buff.getByteBuf());
+                ObjectReader reader = (ObjectReader) invoker.getBindingContext().get(ObjectReader.class.getName());
+                try {
+                    input = reader.readValue((InputStream) in);
+                } catch (Exception e) {
+                    log.error("Failed to unmarshal input", e);
+                    routingContext.fail(400);
+                    return;
                 }
-                Object finalInput = input;
-                executor.execute(() -> {
-                    dispatch(routingContext, invoker, finalInput);
-                });
+            }
+            Object finalInput = input;
+            executor.execute(() -> {
+                dispatch(routingContext, invoker, finalInput);
             });
```

Comment:
It was already there  but given it's runtime code, could we avoid the lambda? (it can be done in a separate PR if you prefer)

Reply:
Definitely - that's an overlook from my side.

Reply:
Fixed.